app/v2/controllers/V2DeparturesController.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.controllers
+
+import akka.stream.Materializer
+import akka.stream.scaladsl.Sink
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import com.google.inject.ImplementedBy
+import com.google.inject.Inject
+import com.google.inject.Singleton
+import play.api.Logging
+import play.api.mvc.Action
+import play.api.mvc.BaseController
+import play.api.mvc.ControllerComponents
+import routing.VersionedRouting
+import v2.controllers.actions.AuthNewEnrolmentOnlyAction
+import v2.controllers.actions.MessageSizeAction
+import v2.controllers.stream.StreamingParsers
+
+import scala.concurrent.Future
+
+@ImplementedBy(classOf[V2DeparturesControllerImpl])
+trait V2DeparturesController {
```

Comment:
Rather than 2 distinct V1, V2 Traits could we have 1 generic as they have the same signature, eg:
 a DepartureController[T] {
def submitDeclaration(): Action[T]
}

Reply:
No - just because `submitDeclration()` might be the same in both, we might end up with routes that will only exist in one or the other controller, or potentially have different parameters - so I'd rather not do that.